### PR TITLE
Move double-click component handler to editor.js

### LINKED ISSUE #969: Double-click-new-glyph code should move from edit-tools-pointer.js to editor.js
The factoring is currently not ideal.

TODO comment:
https://github.com/googlefonts/fontra/blob/1e9a4f6a5fab8134ae6a19630167dc02a66d9f4c/src/fontra/views/editor/edit-tools-pointer.js#L153

The implementation should follow the pattern used by `doubleClickedComponents`.

### PR DESCRIPTION
**Pull request by fatih-erikli**
Fixes #969.

### CODE REVIEW

File: src/fontra/views/editor/edit-tools-pointer.js [hunk 1]
```diff
@@ -149,45 +143,7 @@ export class PointerTool extends BaseTool {
...
-              layers: { [layerName]: { glyph: positionedGlyph.glyph.instance } },
-            })
-          );
-          this.sceneSettings.selectedGlyph = {
-            ...this.sceneSettings.selectedGlyph,
-            isEditing: true,
-          };
-          this.sceneSettings.selectedSourceIndex = 0;
-        }
+        sceneController._dispatchEvent("requestedNewGlyphCreate", { positionedGlyph });
```
Comment (justvanrossum) [inline src/fontra/views/editor/edit-tools-pointer.js:146]: Please do not pass the positionedGlyph to the event, but let the handler get it itself.

File: src/fontra/views/editor/editor.js [hunk 2]
```diff
@@ -332,6 +338,44 @@ export class EditorController {
     });
   }
 
+  async showDialogNewGlyph({ positionedGlyph }) {
```
Comment (justvanrossum) [inline src/fontra/views/editor/editor.js:341]:
```suggestion
  async showDialogNewGlyph() {
      const positionedGlyph = this.sceneController.sceneModel.getSelectedPositionedGlyph();
```
(untested)

Comment (fatih-erikli) [reply src/fontra/views/editor/editor.js]: I tested, works.

File: src/fontra/views/editor/editor.js [hunk 3]
```diff
@@ -177,6 +178,10 @@ export class EditorController {
       this.showDialogGlyphEditLocationNotAtSource();
     });
 
+    this.sceneController.addEventListener("requestedNewGlyphCreate", () => {
```
Comment (justvanrossum) [inline src/fontra/views/editor/editor.js:181]:
```suggestion
    this.sceneController.addEventListener("doubleClickedUndefinedGlyph", () => {
```

File: src/fontra/views/editor/edit-tools-pointer.js [hunk 4]
```diff
@@ -149,45 +143,7 @@ export class PointerTool extends BaseTool {
...
-              layers: { [layerName]: { glyph: positionedGlyph.glyph.instance } },
-            })
-          );
-          this.sceneSettings.selectedGlyph = {
-            ...this.sceneSettings.selectedGlyph,
-            isEditing: true,
-          };
-          this.sceneSettings.selectedSourceIndex = 0;
-        }
+        sceneController._dispatchEvent("requestedNewGlyphCreate");
```
Comment (justvanrossum) [inline src/fontra/views/editor/edit-tools-pointer.js:146]:
```suggestion
        sceneController._dispatchEvent("doubleClickedUndefinedGlyph");
```

Comment (fatih-erikli) [reply src/fontra/views/editor/edit-tools-pointer.js]: I applied them as a single commit

**Review by justvanrossum — APPROVED** [review]

**Pull request merged**